packages/health-check/src/api/index.js

```diff
@@ -2,61 +2,75 @@ const { StatusCodes } = require("http-status-codes");
+ * - OK (200) in case everything is healthy
+ * - SERVICE_UNAVAILABLE (503) in case of any failures or if disabled
+ */
+function getStatusCode() {
+  // check whether the portal has been manually disabled
+  const disabled = getDisabled();
 
   if (disabled) {
     return StatusCodes.SERVICE_UNAVAILABLE;
   }
 
-  // Grab entry element from DB
-  const entry = db.get("entries").orderBy("date", "desc").head().value();
+  // grab one most recent critical entry element from DB
+  const entry = getCurrentCriticalEntry();
 
-  // Check that every critical check entry is up
-  if (entry && entry.checks.every(({ up, critical }) => up && critical)) {
+  // find out whether every check in the entry is up
+  if (entry && entry.checks.every(({ up }) => up)) {
     return StatusCodes.OK;
   }
 
-  // At least one check failed
+  // in case at least one check failed
   return StatusCodes.SERVICE_UNAVAILABLE;
 }
 
-// getTimeout returns the average time out from a sample of 10 health check
-// entries.
-function getTimeout() {
-  if (getStatus() === StatusCodes.SERVICE_UNAVAILABLE) {
-    return 0;
-  }
-
-  // Grab 10 entries from the database as a sample to determine the average
-  // timeout for the server.
+/**
+ * Get the sample of most recent critical entries and
+ * calculate the avarage response time of all of them
+ */
+function getAvarageResponseTime() {
+  // get most recent 10 successfull checks for the calculation
   const sample = db
-    .get("entries")
+    .get("critical")
     .orderBy("date", "desc")
-    .filter(({ checks }) => checks.every(({ up, critical }) => up && critical))
+    .filter(({ checks }) => checks.every(({ up }) => up))
     .take(10)
     .value();
 
-  // Return average timeout
+  // calculate avarage time of response
   return Math.round(sum(sample.map(({ checks }) => sumBy(checks, "time"))) / sample.size);
 }
 
-// getEntriesSinceYesterday gets the health check entries since yesterday
-function getEntriesSinceYesterday() {
-  const yesterday = new Date();
-
-  yesterday.setDate(yesterday.getDate() - 1);
+/**
+ * Get one, most current critical entry
+ */
+function getCurrentCriticalEntry() {
+  return db.get("critical").orderBy("date", "desc").head().value();
+}
 
-  return db
-    .get("entries")
-    .orderBy("date", "desc")
-    .filter(({ date }) => date >= yesterday.toISOString())
-    .value();
+/**
+ * Get the disabled flag state (manual portal disable)
+ */
+function getDisabled() {
+  return db.get("disabled").value();
 }
 
 module.exports = (req, res) => {
+  const statusCode = getStatusCode();
+  const timeout = statusCode === StatusCodes.OK ? getAvarageResponseTime() : 0;
+
+  // We want to delay the response for the load balancer to be able to prioritize
+  // servers based on the successful response time of thid endpoint. Load balancer
+  // will pull the server if the response is an error so there is no point in delaying
+  // failures, hence 0 timeout on those.
```

Comment:
This is a great approach. Also, thank you for the clear and informative comment! I was staring at the `setTimeout` for a while before I read the comment and a lightbulb lit up. :)